Set get for entry hostname

Gustrb opened the pull request:
This Pull Request is a response to the issue: #294
The hostname field is a 255 printable ASCII field in RFC 5424, and users may want to set this to something more meaningful to them. This task implements a pair of functions to set and retrieve the hostname of an entry, which is used if it is set during entry formatting.

Make sure that your pull request follows the guidelines specified in the
[Guidelines for Contributing](../docs/CONTRIBUTING.md).

Commit:
Sfeat(Validate): Add validate functions for hostnames

This functions are the ones responsible for validating the length
of a given hostname, and set the according error codes and responses
according to how it is done in the procid ones

Commit:
feat(Entry): Add methods for setting and getting the hostname

goatshriek commented on include/stumpless/entry.h:
```diff
@@ -1510,18 +1515,72 @@ stumpless_get_entry_procid( const struct stumpless_entry *entry );
+ * appropriately.
+ */
+STUMPLESS_PUBLIC_FUNCTION
+const char *stumpless_get_entry_hostname( const struct stumpless_entry *entry );
```
add a line break after the type, to match the formatting of other declarations

Gustrb replied:
sure!

goatshriek reviewed (changes requested):
Just a few minor changes to make and this should be good to go. You'll need to add entries for your new functions to `src/windows/stumpless.def` to resolve the Windows build issues.

Also, check out the valgrind failure to see if there is a memory leak identified.

Gustrb committed:
fix(Entry): Fix small changes to new hostname api

This adds the two new functions to the stumpless.def file
which allows us to build with those in windows
Also added a new starting value to the struct entry hostname_override
property(false) so that we dont handle with uninitialized memory

goatshriek commented:
Looks like there are still some memory leaks: you will need to free results provided by the get hostname function at a minimum, as this is freshly allocated each time. The valgrind output itself will give you feedback on where the problem is.

Gustrb commented:
how can I check the valgrind output locally??

goatshriek commented:
Assuming you have valgrind installed, you just need to run the test case with it once built. So, from the cmake build directory, something like `valgrind ./function-test-entry` should reproduce the error that the debug build shows.

Gustrb commented:
Hmmm, got it, thank you

Gustrb committed:
fix(Entry): Fixing memory leaks inside the GetHostName tests

goatshriek commented:
Thanks very much for putting this together, and responding to the changes quickly!

Gustrb commented:
I'm really glad to help!!
